apps/aecore/src/aec_pinning_agent.erl

```diff
@@ -20,28 +20,46 @@
     ]).
 
 
+-spec get_pinning_data() -> {ok, map()} | {error, atom()}.
 get_pinning_data() ->
-    {ok, #{epoch  := Epoch,
-           first  := First,
-           last   := _Last,
-           length := _Length
-    }} = aec_chain_hc:epoch_info(),
+    {ok, #{epoch := Epoch,
+           first := First,
+           last  := Last}} = aec_chain_hc:epoch_info(),
+    lager:debug("Get pin data for epoch ~p for leader of block ~p", [Epoch - 1, Last]),
     {ok, BlockHash} = aec_chain_state:get_key_block_hash_at_height(First-1),
     {ok, ChainType} = aec_parent_connector:get_parent_chain_type(),
-    
-    #{epoch             => Epoch-1,
-      height            => First-1,
-      block_hash        => BlockHash,
-      parent_type       => ChainType,
-      parent_network_id => aec_parent_connector:get_network_id()}.
+    ConMod = aec_parent_connector:get_parent_conn_mod(),
+    {_,Type} = lists:split(8,atom_to_list(ConMod)), % split off "aehttpc_" from mod name to get type
```

Comment:
```suggestion
   "aehttpc_" ++ Type = atom_to_list(ConMod),
```

Reply:
or just use the the parent_connector:get_chain_type() implemented in #4430, it´s on `master`

Reply:
Removed this altogether, since now type is known by parent connector.